src/generators/python/renderers/ClassRenderer.ts

```diff
@@ -0,0 +1,97 @@
+import { ClassPresetType } from '../PythonPreset';
+
+/**
+ * Renderer for Python's `class` type
+ * 
+ * @extends PythonRenderer
+ */
+export class ClassRenderer extends PythonRenderer<ConstrainedObjectModel> {
+  async defaultSelf(): Promise<string> {
+    const content = [
+      await this.renderProperties(),
+      await this.runCtorPreset(),
+      await this.renderAccessors(),
+      await this.runAdditionalContentPreset(),
+    ];
+
+    return `class ${this.model.name}: 
+${this.indent(this.renderBlock(content, 2))}
+`;
+  }
+
+  runCtorPreset(): Promise<string> {
+    return this.runPreset('ctor');
+  }
+
+  /**
+   * Render all the properties for the class.
+   */
+  async renderProperties(): Promise<string> {
+    const properties = this.model.properties || {};
+    const content: string[] = [];
+
+    for (const property of Object.values(properties)) {
+      const rendererProperty = await this.runPropertyPreset(property);
+      content.push(rendererProperty);
+    }
+
+    return this.renderBlock(content);
+  }
+
+  runPropertyPreset(property: ConstrainedObjectPropertyModel): Promise<string> {
+    return this.runPreset('property', { property });
+  }
+
+  /**
+   * Render all the accessors for the properties
+   */
+  async renderAccessors(): Promise<string> {
+    const properties = this.model.properties || {};
+    const content: string[] = [];
+
+    for (const property of Object.values(properties)) {
+      const getter = await this.runGetterPreset(property);
+      const setter = await this.runSetterPreset(property);
+      content.push(this.renderBlock([getter, setter]));
+    }
+
+    return this.renderBlock(content, 2);
+  }
+
+  runGetterPreset(property: ConstrainedObjectPropertyModel): Promise<string> {
+    return this.runPreset('getter', { property });
+  }
+
+  runSetterPreset(property: ConstrainedObjectPropertyModel): Promise<string> {
+    return this.runPreset('setter', { property });
+  }
+}
+
+export const PYTHON_DEFAULT_CLASS_PRESET: ClassPresetType<PythonOptions> = {
+  self({ renderer }) {
+    return renderer.defaultSelf();
+  },
+  ctor({renderer, model}) {
+    const properties = model.properties || {};
+    const assigments = Object.values(properties).map((property) => {
+      if (!property.required) {
+        return `if hasattr(input, '${property.propertyName}'):\n\tself._${property.propertyName} = input.${property.propertyName}`;
+      }
+      return `self._${property.propertyName} = input.${property.propertyName}`;
+    });
+    const body = renderer.renderBlock(assigments);
+    return `def __init__(self, input):
+${renderer.indent(body)}`;
+  },
+  getter({ property }) {
+    return `@property
+def ${property.propertyName}(self):\n\treturn self._${property.propertyName}`;
+  },
+  setter({ property }) {
+    return `@${property.propertyName}.setter
+def ${property.propertyName}(self, ${property.propertyName}):\n\tself._${property.propertyName} = ${property.propertyName}`;
+  },
```

Comment:
Should we use decorators as default behaviour? I don't know Python at all, so I only ask.

Reply:
This was what https://github.com/asyncapi/modelina/pull/604 used so I just copied it 🙂 

You will also likely not even reach a consensus as it's preference